plugins/modules/dcnm_service_node.py

```diff
@@ -0,0 +1,741 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import time
+import copy
+import re
+from ansible_collections.cisco.dcnm.plugins.module_utils.network.dcnm.dcnm import get_fabric_inventory_details, \
+    dcnm_send, validate_list_of_dicts, dcnm_get_ip_addr_info, get_ip_sn_dict
+from ansible.module_utils.basic import AnsibleModule
+
+__author__ = "Karthik Babu Harichandra Babu"
+
+DOCUMENTATION = '''
+---
+module: dcnm_service_node
+short_description: Create/Modify/Delete service node based on type and attached interfaces from a DCNM managed VXLAN fabric.
+version_added: "0.9.0"
+description:
+    - "Create/Modify/Delete service node based on type and attached interfaces from a DCNM managed VXLAN fabric."
+author: Karthik Babu Harichandra Babu
+options:
+  fabric:
+    description:
+    - 'Name of attached easy fabric to which service node is attached'
+    type: str
+    required: yes
+  service_fabric:
+    description:
+    - 'Name of external fabric where the service node is located'
+    type: str
+    required: yes
+    description:
+      - The state of DCNM after module completion.
+    type: str
+    choices:
+      - merged
+      - replaced
+      - overridden
+      - deleted
+      - query
+    default: merged
+
+  config:
+    description: 'List of details of service nodes being managed'
+    type: list
+    elements: dict
+    required: true
+    note: Not required for state deleted
+    suboptions:
+      name:
+        description: 'Name of service node'
+        type: str
+        required: true
+      type:
+        description: 'Name of the service node type'
+        type: str
+        required: true
+        default: 'firewall'
+      form_factor:
+        description: 'Name of the form factor of the service node'
+        type: str
+        required: true
+        default: 'physical'
+      svc_int_name:
+        description: 'Name of the service interface'
+        type: str
+        required: true
+      switches:
+        description: 'IP address of the switch where service node will be added/deleted'
+        type: list
+        required: true
+      attach_interface:
+        description: 'List of switch interfaces where the service node will be attached'
+        type: str
+        required: true
+'''
+
+EXAMPLES = '''
+This module supports the following states:
+
+Merged:
+  Service Nodes defined in the playbook will be merged into the service fabric.
+    - If the service node does not exist it will be added.
+    - If the service node exists but properties managed by the playbook are different
+      they will be updated if possible.
+    - Service Nodes that are not specified in the playbook will be untouched.
+
+Replaced:
+  Service Nodes defined in the playbook will be replaced in the service fabric.
+    - If the service node does not exist it will not be added.
```

Comment:
I believe this should read: `If the service node does not exist it will be added.`

Reply:
No mike, in replaced we use the 'PUT' method in replace, if the service node is not available, it will not create, we should be using merged for the same.